### PR TITLE
Update Realm Sync to v4.7.10

### PR DESCRIPTION
**Pull request by kneth**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,8 @@ x.x.x Release notes (yyyy-MM-dd)
 
 ### Internal
 * Cleaned up the console output produced by `./tests`. ([#2548](https://github.com/realm/realm-js/pull/2548))
+* Updated to Realm Sync v4.7.10.
```
Comment (bmunkholm) [inline CHANGELOG.md]:
```suggestion
* Updated to Realm Sync from 4.7.8 to 4.7.10.
```

Comment (bmunkholm) [reply CHANGELOG.md]: Are there no public changes that should be mentioned?

Comment (kneth) [reply CHANGELOG.md]: It is probably worth mentioning that changeset scanning for synced Realm is faster.

Comment (kneth) [reply CHANGELOG.md]: Btw, is your suggestion above a new general "rule" for changelogs?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,10 +1,9 @@
 x.x.x Release notes (yyyy-MM-dd)
 =============================================================
 ### Enhancements
-* None.
+* Improve performance of changeset scanning. For changesets involving large numbers of objects that are otherwise cheap to process this speeds up changeset integration by ~20%. ([realm/realm-sync#3128](https://github.com/realm/realm-sync/pull/3128)).
```
Comment (bmunkholm) [inline CHANGELOG.md]:
```suggestion
* Improve performance of changeset scanning when syncing data. This happens on a background thread, so it shouldn't have any visible effect.
```

**Comment by kneth** [comment]
@kraenhansen Please review (I have added `react-native.config.js` to `package.json`).

**Pull request merged**